SYCL/ESIMD/spec_const/spec_const_ushort.cpp

```diff
@@ -0,0 +1,22 @@
+//==--------------- spec_const_ushort.cpp  - DPC++ ESIMD on-device test ---===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// TODO enable on Windows
```

Comment:
Could you please provide more info about issues on windows?

Reply:
There's no specific error in execution log, except warnings about getenv deprecation ([link](http://134.191.240.166:8010/blue/organizations/jenkins/LLVM-Test-Suite-CI%2FLLVM-Test-Suite-WIN/detail/LLVM-Test-Suite-WIN/156/pipeline/)):

> [2021-02-08T21:31:21.461Z] FAIL: SYCL :: ESIMD/spec_const/spec_const_int.cpp (193 of 430)
> [2021-02-08T21:31:21.461Z] ******************** TEST 'SYCL :: ESIMD/spec_const/spec_const_int.cpp' FAILED ********************
> [2021-02-08T21:31:21.461Z] Script:
> [2021-02-08T21:31:21.461Z] --
> [2021-02-08T21:31:21.461Z] : 'RUN: at line 9';   D:/Jenkins/llvm-test-suite_Jenkins/win_node/workspace/LLVM-Test-Suite-CI/LLVM-Test-Suite-WIN/llvm.obj/bin/clang-cl.exe -fsycl-explicit-simd /EHsc    -fsycl -ID:\Jenkins\llvm-test-suite_Jenkins\win_node\workspace\LLVM-Test-Suite-CI\LLVM-Test-Suite-WIN\llvm-test-suite\SYCL\ESIMD\spec_const/.. D:\Jenkins\llvm-test-suite_Jenkins\win_node\workspace\LLVM-Test-Suite-CI\LLVM-Test-Suite-WIN\llvm-test-suite\SYCL\ESIMD\spec_const\spec_const_int.cpp -o D:\Jenkins\llvm-test-suite_Jenkins\win_node\workspace\LLVM-Test-Suite-CI\LLVM-Test-Suite-WIN\build\SYCL\ESIMD\spec_const\Output\spec_const_int.cpp.tmp.out
> [2021-02-08T21:31:21.461Z] : 'RUN: at line 10';    env SYCL_DEVICE_FILTER=opencl:gpu  D:\Jenkins\llvm-test-suite_Jenkins\win_node\workspace\LLVM-Test-Suite-CI\LLVM-Test-Suite-WIN\build\SYCL\ESIMD\spec_const\Output\spec_const_int.cpp.tmp.out
> [2021-02-08T21:31:21.461Z] --
> [2021-02-08T21:31:21.461Z] Exit Code: 3221226505
> [2021-02-08T21:31:21.461Z] 
> [2021-02-08T21:31:21.461Z] Command Output (stdout):
> [2021-02-08T21:31:21.461Z] --
> [2021-02-08T21:31:21.461Z] $ ":" "RUN: at line 9"
> [2021-02-08T21:31:21.461Z] $ "D:/Jenkins/llvm-test-suite_Jenkins/win_node/workspace/LLVM-Test-Suite-CI/LLVM-Test-Suite-WIN/llvm.obj/bin/clang-cl.exe" "-fsycl-explicit-simd" "/EHsc" "-fsycl" "-ID:\Jenkins\llvm-test-suite_Jenkins\win_node\workspace\LLVM-Test-Suite-CI\LLVM-Test-Suite-WIN\llvm-test-suite\SYCL\ESIMD\spec_const/.." "D:\Jenkins\llvm-test-suite_Jenkins\win_node\workspace\LLVM-Test-Suite-CI\LLVM-Test-Suite-WIN\llvm-test-suite\SYCL\ESIMD\spec_const\spec_const_int.cpp" "-o" "D:\Jenkins\llvm-test-suite_Jenkins\win_node\workspace\LLVM-Test-Suite-CI\LLVM-Test-Suite-WIN\build\SYCL\ESIMD\spec_const\Output\spec_const_int.cpp.tmp.out"
> [2021-02-08T21:31:21.462Z] # command stderr:
> [2021-02-08T21:31:21.462Z] In file included from D:\Jenkins\llvm-test-suite_Jenkins\win_node\workspace\LLVM-Test-Suite-CI\LLVM-Test-Suite-WIN\llvm-test-suite\SYCL\ESIMD\spec_const\spec_const_int.cpp:20:
> [2021-02-08T21:31:21.462Z] In file included from D:\Jenkins\llvm-test-suite_Jenkins\win_node\workspace\LLVM-Test-Suite-CI\LLVM-Test-Suite-WIN\llvm-test-suite\SYCL\ESIMD\spec_const/Inputs/spec_const_common.hpp:12:
> [2021-02-08T21:31:21.462Z] D:\Jenkins\llvm-test-suite_Jenkins\win_node\workspace\LLVM-Test-Suite-CI\LLVM-Test-Suite-WIN\llvm-test-suite\SYCL\ESIMD\spec_const/..\esimd_test_utils.hpp(36,34): warning: 'getenv' is deprecated: This function or variable may be unsafe. Consider using _dupenv_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details. [-Wdeprecated-declarations]
> [2021-02-08T21:31:21.462Z]     if (const char *dev_filter = getenv("SYCL_DEVICE_FILTER")) {
> [2021-02-08T21:31:21.462Z]                                  ^
> [2021-02-08T21:31:21.462Z] C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt\stdlib.h(1190,20): note: 'getenv' has been explicitly marked deprecated here
> [2021-02-08T21:31:21.462Z]     _Check_return_ _CRT_INSECURE_DEPRECATE(_dupenv_s)
> [2021-02-08T21:31:21.462Z]                    ^
> [2021-02-08T21:31:21.462Z] C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\VC\Tools\MSVC\14.16.27023\include\vcruntime.h(255,55): note: expanded from macro '_CRT_INSECURE_DEPRECATE'
> [2021-02-08T21:31:21.462Z]         #define _CRT_INSECURE_DEPRECATE(_Replacement) _CRT_DEPRECATE_TEXT(    \
> [2021-02-08T21:31:21.462Z]                                                       ^
> [2021-02-08T21:31:21.462Z] C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\VC\Tools\MSVC\14.16.27023\include\vcruntime.h(245,47): note: expanded from macro '_CRT_DEPRECATE_TEXT'
> [2021-02-08T21:31:21.462Z] #define _CRT_DEPRECATE_TEXT(_Text) __declspec(deprecated(_Text))
> [2021-02-08T21:31:21.462Z]                                               ^
> [2021-02-08T21:31:21.462Z] 1 warning generated.
> [2021-02-08T21:31:21.462Z] In file included from D:\Jenkins\llvm-test-suite_Jenkins\win_node\workspace\LLVM-Test-Suite-CI\LLVM-Test-Suite-WIN\llvm-test-suite\SYCL\ESIMD\spec_const\spec_const_int.cpp:20:
> [2021-02-08T21:31:21.462Z] In file included from D:\Jenkins\llvm-test-suite_Jenkins\win_node\workspace\LLVM-Test-Suite-CI\LLVM-Test-Suite-WIN\llvm-test-suite\SYCL\ESIMD\spec_const/Inputs/spec_const_common.hpp:12:
> [2021-02-08T21:31:21.462Z] D:\Jenkins\llvm-test-suite_Jenkins\win_node\workspace\LLVM-Test-Suite-CI\LLVM-Test-Suite-WIN\llvm-test-suite\SYCL\ESIMD\spec_const/..\esimd_test_utils.hpp(36,34): warning: 'getenv' is deprecated: This function or variable may be unsafe. Consider using _dupenv_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details. [-Wdeprecated-declarations]
> [2021-02-08T21:31:21.462Z]     if (const char *dev_filter = getenv("SYCL_DEVICE_FILTER")) {
> [2021-02-08T21:31:21.462Z]                                  ^
> [2021-02-08T21:31:21.462Z] C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt\stdlib.h(1190,20): note: 'getenv' has been explicitly marked deprecated here
> [2021-02-08T21:31:21.462Z]     _Check_return_ _CRT_INSECURE_DEPRECATE(_dupenv_s)
> [2021-02-08T21:31:21.462Z]                    ^
> [2021-02-08T21:31:21.462Z] C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\VC\Tools\MSVC\14.16.27023\include\vcruntime.h(255,55): note: expanded from macro '_CRT_INSECURE_DEPRECATE'
> [2021-02-08T21:31:21.462Z]         #define _CRT_INSECURE_DEPRECATE(_Replacement) _CRT_DEPRECATE_TEXT(    \
> [2021-02-08T21:31:21.462Z]                                                       ^
> [2021-02-08T21:31:21.462Z] C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\VC\Tools\MSVC\14.16.27023\include\vcruntime.h(245,47): note: expanded from macro '_CRT_DEPRECATE_TEXT'
> [2021-02-08T21:31:21.462Z] #define _CRT_DEPRECATE_TEXT(_Text) __declspec(deprecated(_Text))
> [2021-02-08T21:31:21.462Z]                                               ^
> [2021-02-08T21:31:21.462Z] 1 warning generated.
> [2021-02-08T21:31:21.462Z] In file included from D:\Jenkins\llvm-test-suite_Jenkins\win_node\workspace\LLVM-Test-Suite-CI\LLVM-Test-Suite-WIN\llvm-test-suite\SYCL\ESIMD\spec_const\spec_const_int.cpp:20:
> [2021-02-08T21:31:21.462Z] In file included from D:\Jenkins\llvm-test-suite_Jenkins\win_node\workspace\LLVM-Test-Suite-CI\LLVM-Test-Suite-WIN\llvm-test-suite\SYCL\ESIMD\spec_const/Inputs/spec_const_common.hpp:12:
> [2021-02-08T21:31:21.462Z] D:\Jenkins\llvm-test-suite_Jenkins\win_node\workspace\LLVM-Test-Suite-CI\LLVM-Test-Suite-WIN\llvm-test-suite\SYCL\ESIMD\spec_const/..\esimd_test_utils.hpp(36,34): warning: 'getenv' is deprecated: This function or variable may be unsafe. Consider using _dupenv_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details. [-Wdeprecated-declarations]
> [2021-02-08T21:31:21.462Z]     if (const char *dev_filter = getenv("SYCL_DEVICE_FILTER")) {
> [2021-02-08T21:31:21.463Z]                                  ^
> [2021-02-08T21:31:21.463Z] C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt\stdlib.h(1190,20): note: 'getenv' has been explicitly marked deprecated here
> [2021-02-08T21:31:21.463Z]     _Check_return_ _CRT_INSECURE_DEPRECATE(_dupenv_s)
> [2021-02-08T21:31:21.463Z]                    ^
> [2021-02-08T21:31:21.463Z] C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\VC\Tools\MSVC\14.16.27023\include\vcruntime.h(255,55): note: expanded from macro '_CRT_INSECURE_DEPRECATE'
> [2021-02-08T21:31:21.463Z]         #define _CRT_INSECURE_DEPRECATE(_Replacement) _CRT_DEPRECATE_TEXT(    \
> [2021-02-08T21:31:21.463Z]                                                       ^
> [2021-02-08T21:31:21.463Z] C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\VC\Tools\MSVC\14.16.27023\include\vcruntime.h(245,47): note: expanded from macro '_CRT_DEPRECATE_TEXT'
> [2021-02-08T21:31:21.463Z] #define _CRT_DEPRECATE_TEXT(_Text) __declspec(deprecated(_Text))
> [2021-02-08T21:31:21.463Z]                                               ^
> [2021-02-08T21:31:21.463Z] 1 warning generated.
> [2021-02-08T21:31:21.463Z] 
> [2021-02-08T21:31:21.463Z] $ ":" "RUN: at line 10"
> [2021-02-08T21:31:21.463Z] $ "env" "SYCL_DEVICE_FILTER=opencl:gpu" "D:\Jenkins\llvm-test-suite_Jenkins\win_node\workspace\LLVM-Test-Suite-CI\LLVM-Test-Suite-WIN\build\SYCL\ESIMD\spec_const\Output\spec_const_int.cpp.tmp.out"
> [2021-02-08T21:31:21.463Z] note: command had no output on stdout or stderr
> [2021-02-08T21:31:21.463Z] error: command failed with exit status: 3221226505

Reply:
The log looks like the case when unhandled exception if thrown from application.

Could you please add SYCL exception handling to the test: see examples in https://github.com/intel/llvm-test-suite/pull/63
to get extra information about failure.